### DISCUSSION TITLE
How to disable parameter in Swagger?

### DISCUSSION THREAD

**Discussion by nickshanks347** [fastapi/fastapi#5092] [Questions]
I'm using a function that looks like this:

```python
@router.get("/cookie")
def get_cookie(current_user: User = Depends(AuthCore.get_current_user), access_token: str = Cookie(default=None)):
    return {"current_user": current_user.username, "access_token": access_token}
```

You can see that the route needs `access_token` and this appears in swagger like so:
![image](https://user-images.githubusercontent.com/11299982/176575102-1e584e20-1766-4913-9290-4446a3acd850.png)

As the route gets a cookie called `access_token` there is no way for the user to edit the cookie, so, is there any way to disable the `access_token` parameter in Swagger while still actually calling the `access_token` cookie on the `get_cookie` function?

**Comment by frankie567** [comment]
Use the [`include_in_schema` parameter](https://fastapi.tiangolo.com/tutorial/query-params-str-validations/#exclude-from-openapi):

```py
@router.get("/cookie")
def get_cookie(current_user: User = Depends(AuthCore.get_current_user), access_token: str = Cookie(default=None, include_in_schema=False)):
    return {"current_user": current_user.username, "access_token": access_token}
```

**Reply by YuriiMotov** [reply]
It will not only remove this parameter from Swagger, but from openapi schema.
So, the schema will be actually incorrect

I think it would be better to try configuring Swagger interface to disable input fields for cookie parameters. Not sure there is such config option
